xwiki-rendering-api/src/main/java/org/xwiki/rendering/util/IconProvider.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.util;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Component to provide an icon from its name.
+ *
+ * @version $Id$
+ * @since 16.7.0RC1
+ */
+@Role
+@Unstable
+public interface IconProvider
+{
+    /**
+     * Provides an icon from its name.
+     * @param iconName the name of the icon needed
+     * @return the block containing the icon
+     */
+    Block get(String iconName);
```

Comment:
This always returns inline content ? If yes, it should be more explicit in the API.

Reply:
At the moment, the implementation in xwiki-platform always returns a raw block with HTML but as we allow defining icons with wiki syntax in Icon themes, this could be any block that can be generated with XWiki syntax.

Reply:
Sure, my point was mostly about improving the javadoc to give examples (usually examples are the best to explain an api)

Reply:
> This always returns inline content ? If yes, it should be more explicit in the API documentation.

It would make sense to have this limitation, but so far we didn't need it to be specifically inline so I don't think we need to set a comment about it in the API. If at some point we need a function that returns only inline content, we can always extend the API and add `getInline`.

> I think you should provide some examples in the javadoc about types of Block that can be used: ImageBlock (icons represented by images) + the Block type used for FA.

:+1: done in https://github.com/xwiki/xwiki-rendering/pull/296/commits/c03fabe496852a0f1d5cea38b8c52d83ca3e539d

Reply:
Added in  https://github.com/xwiki/xwiki-rendering/pull/296/commits/055972f22dff4437e94aeefeafce54953cce957f that this method always returns an inline icon.